Frontend/ui-customer/src/Flow.purs

```diff
@@ -1449,6 +1475,246 @@ helpAndSupportScreenFlow = do
+      )issueInfoRes.chats)
+      let categoryName = getTitle issueInfoRes.categoryLabel
+      let showStillHaveIssue' = case (last issueInfoRes.chats) of
+                      Just (ChatDetail msg) -> (fromMaybe "" msg.label) == "AUTO_MARKED_RESOLVED"
+                      Nothing -> false
+      modifyScreenState $ ReportIssueChatScreenStateType (\_ -> ReportIssueChatScreenData.initData { data {showStillHaveIssue = showStillHaveIssue', categoryId = issueInfoRes.categoryId, categoryName = categoryName, options = options', issueId = Just selectedIssue.issueReportId, chatConfig = ReportIssueChatScreenData.initData.data.chatConfig{messages = messages'}, isResolved = selectedIssue.status == "CLOSED" }, props {showSubmitComp = false}})
+      issueReportChatScreenFlow
+
+issueReportChatScreenFlow :: FlowBT String Unit
+issueReportChatScreenFlow = do
+  modifyScreenState $ ReportIssueChatScreenStateType (\state -> state { props {isKeyboardOpen = false}})
+  flow <- UI.reportIssueChatScreen
+  case flow of
+    SELECT_ISSUE_OPTION updatedState -> do
+      let selectedOptionId = fromMaybe "" (map (\option -> option.issueOptionId) updatedState.data.selectedOption)
+      let selectedOptionLabel = fromMaybe "" (map (\option -> option.label) updatedState.data.selectedOption)
+      let language = fetchLanguage $ getValueToLocalStore LANGUAGE_KEY
+      let isResolved = selectedOptionLabel == "MARK_RESOLVED"
+      (GetOptionsRes getOptionsRes) <- Remote.getOptionsBT language updatedState.data.categoryId selectedOptionId (fromMaybe "" updatedState.data.issueId)
+      when isResolved do
+        let updateIssueReqBody = UpdateIssueReqBody {status : "CLOSED"}
+        (UpdateIssueRes _) <- Remote.updateIssue (fromMaybe "" updatedState.data.issueId) language updateIssueReqBody
+        pure unit
+      let getOptionsRes' = (mapWithIndex (\index (Option x) ->
+        { option : (show (index + 1)) <> ". " <>
+                   if (language == "en")
+                    then capitalizeFirstChar x.option
+                    else x.option
+        , issueOptionId : x.issueOptionId
+        , label : x.label
+        }
+      ) getOptionsRes.options)
+      let messages' = (mapWithIndex (\index (Message currMessage) -> 
+        (makeChatComponent' currMessage.message "Bot" (getCurrentUTC "") "Text" (500 * (index + 1)))
+      )getOptionsRes.messages)
+      let chats' = ([Chat {
+          chatId : selectedOptionId,
+          chatType : "IssueOption",
+          timestamp : (getCurrentUTC "")
+        }] <> (map (\(Message currMessage) -> Chat {
+          chatId : currMessage.id,
+          chatType : "IssueMessage",
+          timestamp : (getCurrentUTC "")
+        }
+      )getOptionsRes.messages))
+      let showSubmitComp = any (\ (Message  x) -> (fromMaybe "" x.label) == "CREATE_TICKET") getOptionsRes.messages
```

Comment:
can we refactor code to remove extra brackets and redundant `let` other places also